pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -778,6 +786,44 @@ func validateNamedRuleWithOperations(n *admissionregistration.NamedRuleWithOpera
 	return allErrors
 }
 
+func validateMatchCondition(v *admissionregistration.MatchCondition, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	trimmedExpression := strings.TrimSpace(v.Expression)
+	if len(trimmedExpression) == 0 {
+		allErrors = append(allErrors, field.Required(fldPath.Child("expression"), "expression is not specified"))
+	} else {
+		result := plugincel.CompileCELExpression(
+			&generic.MatchCondition{
+				Expression: trimmedExpression,
+			},
+			plugincel.OptionalVariableDeclarations{
+				HasParams:     false,
+				HasAuthorizer: false,
+			},
+			celconfig.PerCallLimit)
+		if result.Error != nil {
+			switch result.Error.Type {
+			case cel.ErrorTypeRequired:
+				allErrors = append(allErrors, field.Required(fldPath.Child("expression"), result.Error.Detail))
+			case cel.ErrorTypeInvalid:
+				allErrors = append(allErrors, field.Invalid(fldPath.Child("expression"), v.Expression, result.Error.Detail))
+			case cel.ErrorTypeInternal:
+				allErrors = append(allErrors, field.InternalError(fldPath.Child("expression"), result.Error))
+			default:
+				allErrors = append(allErrors, field.InternalError(fldPath.Child("expression"), fmt.Errorf("unsupported error type: %w", result.Error)))
+			}
+		}
+	}
+	if len(v.Name) == 0 {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("name"), v.Name, "name must be specified"))
+	}
+	nameErrors := utilvalidation.IsQualifiedName(v.Name)
+	for _, nameError := range nameErrors {
+		allErrors = append(allErrors, field.Invalid(fldPath.Child("name"), v.Name, nameError))
+	}
```

Comment:
Also, document in the API docs that `name` must be unique for each `matchCriteria` and that it must be a qualified name?

Reply:
used `apivalidation.ValidateQualifiedName` but we're not currrently validating that each name is unique for each matchCriteria? Should I add that?

Reply:
Yes, see https://github.com/kubernetes/kubernetes/pull/116261#discussion_r1130422437. Unfortunately, "+listType:map" does not auto-validate this (maybe in the future it will).

Reply:
Okay I can add that unique name check. 

Also had to use `utilvalidation.IsQualifiedName` the webhook name validator also uses the utilvalidation way. apivalidation wasn't behaving but this seems to work